.gitignore

```diff
@@ -131,3 +131,6 @@ venv.bak/
 dmypy.json
 
 *.cproject
+
+# generated parser
+src/scenic/parser/parser.py
```

Comment:
Any particular reason to make new folders instead of putting both the grammar and the generated parser under `src/syntax`?

Reply:
I simply followed the structure of the cpython repository where `/Grammar` is just for storing the `.gram` file. If `src` is for storing Python files, maybe it makes sense to have a separate directory for non-Python sources? I don't have a strong opinion on this, so I'm fine with putting the grammar inside `src/syntax` or maybe `src/parser`.

Reply:
63621cf9d1c4e14b4c8649eee1000a1499348e91

moved both the grammar and generated parser under `syntax`